src/components/registration-form/ParentEnrolledConfirmationPage.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import { Center, Box, Text, Button, VStack } from "@chakra-ui/react";
+import ApprovedIcon from "@components/icons/ApprovedIcon";
+import colourTheme from "@styles/colours";
+import Link from "next/link";
+import { createClassRegistration } from "@utils/createClassRegistration";
+import { Student } from "@prisma/client";
+import { Loading } from "@components/Loading";
+
+type ParentEnrolledConfirmationPageProps = {
+    student: Student;
+    classId: number;
+};
+
+export const ParentEnrolledConfirmationPage: React.FC<ParentEnrolledConfirmationPageProps> =
+    ({ student, classId }): JSX.Element => {
+        const data = createClassRegistration(student, classId);
+
+        if (!data) return <Loading></Loading>;
```

Comment:
I think I this should fix the double POST call issue. Always expect a page to render more than once, so we should always place safeguards.
```suggestion
        const [success, setSuccess] = useState(false);
        let data;

        if (!success) {
            data = createClassRegistration(student, classId).then((res) => res);
            setSuccess(true);
        }

        if (data === null) return <Loading />;
```